internal/framework6provider/float64_precision_resource_test.go

```diff
@@ -72,3 +75,161 @@ func TestSchemaResource_Float64Attribute_Precision_Plan(t *testing.T) {
 		},
 	})
 }
+
+// This test documents edge case behavior with math.MaxFloat32
+// This number is unique in that it has an exact representation in both float32 and float64,
+// it is also an integer, which may affect how this value gets encoded/decoded with go-cty.
+// https://github.com/hashicorp/terraform-plugin-framework/issues/1017
+func TestSchemaResource_Float64Attribute_Precision_MaxFloat32(t *testing.T) {
+	resource.UnitTest(t, resource.TestCase{
+		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+			// go-cty v1.14.4 uses string msgpack encoding instead of float msgpack encoding for large whole numbers
+			// https://github.com/hashicorp/terraform/commit/cd252557e4200e031b942d9ff35c455bb30d858f
```

Comment:
Nit: Might be a better breadcrumb to link the PR: https://github.com/hashicorp/terraform/pull/34756, which has some useful notes + links to the go-cty PR

Similar note for the other tests and `framework5provider` equivalents.